deps/build.jl

```diff
@@ -1,58 +1,93 @@
+#     "https://github.com/JuliaOpt/COINBLASBuilder/releases/download/v1.4.6-1/build_COINBLASBuilder.v1.4.6.jl",
+#     "https://github.com/JuliaOpt/ASLBuilder/releases/download/v3.1.0-1/build_ASLBuilder.v3.1.0.jl"
+]
+                  
+custom_library = false
+if haskey(ENV,"JULIA_CLP_LIBRARY_PATH")
+    custom_products = [LibraryProduct(ENV["JULIA_CLP_LIBRARY_PATH"],product.libnames,product.variable_name) for product in products]
+    if all(satisfied(p; verbose=verbose) for p in custom_products)
+        products = custom_products
+        custom_library = true
+    else
+        error("Could not install custom libraries from $(ENV["JULIA_CLP_LIBRARY_PATH"]).\nTo fall back to BinaryProvider call delete!(ENV,\"JULIA_CLP_LIBRARY_PATH\") and run build again.")
+    end
+end                    
 
-# Install unsatisfied or updated dependencies:                    
-unsatisfied = any(!satisfied(p; verbose=verbose) for p in products)
-if haskey(download_info, platform_key())
-    url, tarball_hash = download_info[platform_key()]
-    # Check if this build.jl is providing new versions of the binaries, and
-    # if so, ovewrite the current binaries even if they were installed by the user 
-    if unsatisfied || !isinstalled(url, tarball_hash; prefix=prefix) 
-        # Download and install binaries
-        for dependency in reverse(dependencies)          # We do not check for already installed dependencies
-            download(dependency,basename(dependency))
-            evalfile(basename(dependency))
-        end                            
-        install(url, tarball_hash; prefix=prefix, force=true, verbose=verbose)
+if !custom_library
+    # Install unsatisfied or updated dependencies:
+    unsatisfied = any(!satisfied(p; verbose=verbose) for p in products)
+            
+    dl_info = choose_download(download_info, this_platform)
+    if dl_info === nothing && unsatisfied
+        # If we don't have a compatible .tar.gz to download, complain.
+        # Alternatively, you could attempt to install from a separate provider,
+        # build from source or something even more ambitious here.
+        error("Your platform (\"$(Sys.MACHINE)\", parsed as \"$(triplet(platform_key_abi()))\") is not supported by this package!")
     end
-elseif unsatisfied
-    # If we don't have a BinaryProvider-compatible .tar.gz to download, complain.
-    # Alternatively, you could attempt to install from a separate provider,
-    # build from source or something even more ambitious here.
-    error("Your platform $(triplet(platform_key())) is not supported by this package!")
-end
 
+    # If we have a download, and we are unsatisfied (or the version we're
+    # trying to install is not itself installed) then load it up!
+    if unsatisfied || !isinstalled(dl_info...; prefix=prefix)
+        # Download and install binaries
+    # no dynamic dependencies until Pkg3 support for binaries
```

Comment:
Inconsistent indent level of the comments. Usually comments start at the indent level of the surrounding code, e.g.,
```julia
if foo
    f()
    # This is a comment.
end
```